### PR TITLE
check for all valid CSS font size units

### LINKED ISSUE #5337: Charts and plots not rendering with user defined title text_font_size in em
As the title states, If I add a defined font size to a title on a plot or a chart, that plot and/or chart does not render in Bokeh 0.12.3. This is a pretty large issue. Reproducible below:

```
#!/usr/bin/python
# -*- coding: utf-8 -*-
from bokeh import plotting
from bokeh.models import ColumnDataSource
from bokeh.io import curdoc, show
import pandas as pd

data = [{"1": 10, "2": 4, "3": 3,
         "4": 5, "5": 15, "6": 7,
         "7": 5, "8": 10}]

source = pd.DataFrame(data)
source = ColumnDataSource(source)

fig = plotting.figure(width=1200, height=900, title="BL-55-T55555")

fig.vbar(top="1", x=1, width=0.5, fill_color="blue", line_color="black", source=source)
fig.vbar(top="2", x=2, width=0.5, fill_color="green", line_color="black", source=source)
fig.vbar(top="3", x=3, width=0.5, fill_color="yellow", line_color="black", source=source)
fig.vbar(top="4", x=4, width=0.5, fill_color="orange", line_color="black", source=source)
fig.vbar(top="5", x=5, width=0.5, fill_color="red", line_color="black", source=source)
fig.vbar(top="6", x=6, width=0.5, fill_color="purple", line_color="black", source=source)
fig.vbar(top="7", x=7, width=0.5, fill_color="grey", line_color="black", source=source)
fig.vbar(top="8", x=8, width=0.5, color="black", source=source)

fig.title.text_font_size = '1.9em'

curdoc().add_root(fig)

show(fig)
```

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #5337
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

Took list of units from here:

http://www.w3schools.com/cssref/css_units.asp

Tested every unit by hand to see that:

* it rendered
* it changed size when the value changed

### CODE REVIEW

**Comment by bryevdv** [comment]
@bokeh/dev this is ready for review

**Comment by canavandl** [comment]
LGTM - a crude for loop shows that all font units are canvas supported.

```
from bokeh.models import Label
from bokeh.plotting import figure, show

p = figure(x_range=(0,25), y_range=(0,25))

for i, unit in enumerate("%%|rem|vh|vw|ch|em|ex|cm|mm|in|px|pt|pc".split("|")):
    font_size = "5{}".format(unit)
    x = y = i * 1.5
    p.add_layout(Label(x=x, y=y, text=font_size, text_font_size=font_size))

show(p)
```

screen shot 2016-10-12 at 5 02 14 pm

**Comment by bryevdv** [comment]
Hrm I think it should actually just be `%` not `%%` I was thinking it had to be escaped but that's only when used in a format string.

**Comment by canavandl** [comment]
that fixed the "%" unit

screen shot 2016-10-12 at 5 30 12 pm

File: bokeh/core/properties.py [hunk 1]
```diff
@@ -1795,8 +1795,17 @@ class FontSizeSpec(DataSpec):
...
+    interpreted as a CSS length, then the DataSpec is a value spec.
+    Otherwise, setting a string value will result in a field spec.
+
+    A full list of all valid CSS length units can be found here:
+
+    http://www.w3schools.com/cssref/css_units.asp
+
     '''
-    _font_size_re = re.compile("^[0-9]+(\.[0-9]+)?(ex|px|cm|mm|in|pt|pc)$")
+    _font_size_re = re.compile("^[0-9]+(\.[0-9]+)?(%|rem|vh|vw|ch|em|ex|cm|mm|in|px|pt|pc)$")
```
Comment (birdsarah) [inline bokeh/core/properties.py]: It would be helpful if these were listed in alphabetical order.

**Comment by birdsarah** [comment]
Found this reference too in case it helps:

https://developer.mozilla.org/en-US/docs/Web/CSS/length

**Comment by bryevdv** [comment]
I think I prefer the `w3schools` link, tho I suppose the definitive source would behttps://drafts.csswg.org/css-values/#lengths

There are some additional units in there but they are awfully obscure. It did make me question, is capitalization important? your link has `q` but the CSSWG link has `Q`

**Comment by birdsarah** [comment]
I think the CSS wg would be better.

Regarding case: https://drafts.csswg.org/css-values/#dimensions

Like keywords, unit identifiers are ASCII case-insensitive.

(i.e., [a-z] and [A-Z] are equivalent).

**Comment by bryevdv** [comment]
Ok I will update the PR later tonight

**Comment by bryevdv** [comment]
@birdsarah OK I updated to not be case sensitive. I put the units in the same order as the csswg tables, which is the most useful order for me.

**Comment by birdsarah** [comment]
Cool!

**Pull request merged**